jax_unirep/evotuning.py

```diff
@@ -251,38 +258,90 @@ def step(i, state):
 
         return state
 
+    # Load and check that params have correct keys and shapes
+    if params is None:
+        params = load_params()
+
+    # Defensive programming checks
+    if len(params) != len(model_layers):
+        raise ValueError(
+            "The number of parameters specified must match the number of stax.serial layers"
+        )
+    validate_mLSTM1900_params(params[0])
+
+    if batch_method not in ["length", "random"]:
+        raise ValueError("batch_method must be one of 'length' or 'random'")
+
+    if batch_method == "length":
+        # batch sequences by length
+        xs, ys, seq_lens = length_batch_input_outputs(sequences)
+        len_batching_funcs = {
+            k: get_batching_func(x, y, batch_size)
+            for (k, x, y) in zip(seq_lens, xs, ys)
+        }
+
+        batch_lens = [len(batch) for batch in xs]
+        logger.info(
+            f"Length-batching done: "
+            f"{len(batch_lens)} unique lengths, "
+            f"with average length {onp.mean(batch_lens)}, "
+            f"max length {max(batch_lens)} and min length {min(batch_lens)}."
+        )
```

Comment:
This is maybe more confusing than helpful now, since those batches will get subsampled in training.